pointblankdev/tailwindui-storybook
# Update script to support current tailwindui.com

**What**
Update scraper to make it work for the new tailwindui.com site. This PR also tries to improve the codebase so it's easier to maintain.

Checklist:
- [x] Add `.env.sample` as a template for environment variables required by the script
- [x] Add npm scripts for entry point (eg. `yarn start react`, `yarn create-mdx react`)
- [x] Add screenshots in docs/hints for future guidance
- [x] Extract codes into util/ and lib/
- [x] Add guest mode (run the script site without login info) to scrape free components
- [x] Update playwright

**Why**
The previous code no longer work due to changes in tailwindui.com.

**Test plan**

- Copy .env.sample into .env
- Add your TailwindUI login info to the .env file
- Run `yarn start react` and it should create a json file at `./output/tailwindui.react.json`
- Run `yarn create-mdx react` and it should create the Storybook stories at `./output/tailwindui-react/`

Preview:

https://user-images.githubusercontent.com/7823011/199366668-89ea07ba-0551-4fa5-8984-ba5d81cde9c1.mp4

I also tested the generated code in [Stackblitz](https://stackblitz.com/edit/twui-storybook?file=src%2Fcomponents%2FHeroSections%2Fstories%2Findex.stories.mdx) to ensure that it works with current latest Storybook. Screenshot:

<img src=https://user-images.githubusercontent.com/7823011/199364823-e61e7deb-3db1-453d-9202-45240cefb0c2.png />
